### PR TITLE
Reformating of Strategix and Cetautomatix nodes

### PR DESCRIPTION
**Pull request by mAxYoLo01**
## Purpose

The goal of this PR is to reformat most of Cetautomatix and Strategix nodes, as they were poorly written in the previous year

## Details of what has changed
- Created an Action class as well as all its children to represent all possible types of Actions. This allows for better customization and much better comprehension of the entire codebase
- Temporarily removed most actions, to put them back one by one once they are done. Currently present: Phare & Pavillon
- Merged Score class with the Strategix node (useless class on its own)
- **BIG** change: New Robot class with much more readable functions (seeing the difference between base and custom functions), trying to separate some actions in their own class. Old Robot class still present in the old_robot.py for the moment.

### CODE REVIEW

File: src/assurancetourix/strategix/strategix/actions.py [hunk 1]
```diff
@@ -1,92 +1,104 @@
...
+#     "GOB30": {"COLOR": "GREEN", "IN_ECUEIL": True, "ONLY_SIDE": "yellow"},
+#     "GOB32": {"COLOR": "GREEN", "IN_ECUEIL": True, "ONLY_SIDE": "yellow"},
+#     "GOB34": {"COLOR": "GREEN", "IN_ECUEIL": True, "ONLY_SIDE": "yellow"},
+#     # Cups in ECUEIL_1 & ECUEIL_2 following the Scenario 1:
+#     "GOB35": {"COLOR": "GREEN", "IN_ECUEIL": True},
+#     "GOB37": {"COLOR": "GREEN", "IN_ECUEIL": True},
+#     "GOB38": {"COLOR": "GREEN", "IN_ECUEIL": True},
+#     "GOB40": {"COLOR": "GREEN", "IN_ECUEIL": True},
+#     "GOB43": {"COLOR": "GREEN", "IN_ECUEIL": True},
+# }
```
Comment (3wnbr1) [inline src/assurancetourix/strategix/strategix/actions.py:104]: Why ?! 😂🆘

File: src/assurancetourix/strategix/strategix/action_objects/action.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
+class Action:
```
Comment (3wnbr1) [inline src/assurancetourix/strategix/strategix/action_objects/action.py:1]: Add a docstring to document that this is an abstract base class

File: src/assurancetourix/strategix/strategix/action_objects/manche_air.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
+from .action import Action
+
+
+class MancheAir(Action):
+    def __init__(self, position, **kwargs):
+        super().__init__(position, **kwargs)
+        self.step = 0
+
+    def get_initial_position(self, robot):
```
Comment (3wnbr1) [inline src/assurancetourix/strategix/strategix/action_objects/manche_air.py:9]: Add docstring

File: src/assurancetourix/strategix/strategix/action_objects/manche_air.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+        if self.step == 0:
+            offset = -0.1 if self.tags["ONLY_SIDE"] == "blue" else 0.1
+        else:
+            offset = 0.1 if self.tags["ONLY_SIDE"] == "blue" else -0.1
+        position = (self.position[0] + offset, robot.width_param / 2 + 0.05)
+        return position
+
+    def start_actuator(self, robot):
```
Comment (3wnbr1) [inline src/assurancetourix/strategix/strategix/action_objects/manche_air.py:17]: Same

File: src/assurancetourix/strategix/strategix/strategix.py [hunk 5]
```diff
@@ -103,6 +101,72 @@ def update_score(self):
...
+        # if self.bonPortGros.pos == self.bonPortPetit.pos == 'Good':
+        #     self.score += 10
+        # elif self.bonPortGros.pos == self.bonPortPetit.pos == 'Wrong':
+        #     self.score += 5
+        # elif self.bonPortGros.pos == 'Good' and self.bonPortPetit.pos == 'Out':
+        #     self.score += 5
+        # elif self.bonPortGros.pos == 'Out' and self.bonPortPetit.pos == 'Good':
+        #     self.score += 5
+        # else:
+        #     self.score += 0
```
Comment (3wnbr1) [inline src/assurancetourix/strategix/strategix/strategix.py:134]: Again ?

File: src/modules/cetautomatix/cetautomatix/old_robot.py [hunk 6]
```diff
@@ -0,0 +1,505 @@
...
+        #             if pump_dict.get('type') == NO and pump_dict.get('STATUS') is not None:
+        #                 arriere = True
+        #         if arriere is True:
+        #             offset = (0, -0.1)
+        #             elements[self._current_action]["Rot"] = -90
+        #             self.drop = NO
+        #         else:
+        #             offset = (0, 0.1)
+        #             elements[self._current_action]["Rot"] = 90
+        #             self.drop = NC
```
Comment (3wnbr1) [inline src/modules/cetautomatix/cetautomatix/old_robot.py:456]: No

**Review by PhileasL — APPROVED** [review]
Approve for the moment, I'll review it properly later. @mAxYoLo01 can continue his work this afternoon.

File: src/modules/cetautomatix/cetautomatix/old_robot.py [hunk 7]
```diff
@@ -0,0 +1,505 @@
+#!/usr/bin/env python3
```
Comment (3wnbr1) [inline src/modules/cetautomatix/cetautomatix/old_robot.py:1]: Drop the file if you don't use it anymore. Git is there for you

**Review by 3wnbr1 — CHANGES_REQUESTED** [review]
Nice effort 👍

**Review by 3wnbr1 — APPROVED** [review]
Congratz

**Pull request merged**